src/App.tsx

```diff
@@ -41,20 +41,43 @@ async function getScannedCards(dlens: FileList, sqlite3: any, db: any) {
 // Returns a mapping of card IDs to card names
 async function getApkCardTable(apk: FileList, sqlite3: any, db: any) {
 
-  const apkBuffer = await (await ZipArchive.from_blob(apk[0]))
-    .get("res/raw/data.db")
-    ?.get_array_buffer();
+  const apkArchive = await ZipArchive.from_blob(apk[0]);
+  const iterator:Iterator<[file_name: string, entry: ZipEntry]> = apkArchive.files();
+  let dbFileEntry:any = null;
 
-  if (!apkBuffer) {
-    throw new Error("Couldn't read APK");
+  let currentFile:IteratorResult<[file_name: string, entry: ZipEntry], any> = iterator.next();
+
+  while(!currentFile.done && currentFile !== null)
+  {
+    const filename:string = currentFile.value[0];
+
+    if(filename.endsWith(".db")) 
+    {
+      dbFileEntry = currentFile.value[1];
+      break;
+    }
+    currentFile = iterator.next();
+  }
+
+  dbFileEntry = false;
+
+  if (!dbFileEntry) {
+    alert("The provided APK file did not contain any *.db file");
```

Comment:
I would rather this be done with error boundaries, but I'm happy to do that myself after this is merged.